presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -123,8 +126,15 @@ public static ActualProperties deriveProperties(PlanNode node, List<ActualProper
         verify(node.getOutputSymbols().containsAll(localPropertyColumns), "Node-level local properties contain columns not present in node's output");
 
         // TODO: ideally this logic would be somehow moved to PlanSanityChecker
-        verify(node instanceof SemiJoinNode || inputProperties.stream().noneMatch(ActualProperties::isNullsReplicated) || output.isNullsReplicated(),
-                "SemiJoinNode is the only node that can strip null replication");
+        if (!(node instanceof SemiJoinNode)) {
+            Set<Replication> inputReplications = inputProperties.stream()
+                    .map(ActualProperties::getReplication)
+                    .filter(replication -> !replication.isNoReplication())
```

Comment:
`.filter(Replication::replicates)` is less verbose than:
`.filter(replication -> !replication.isNoReplication())`.
Other places don't seem to be affected either way.
It is also mentally easier to think in terms of direct logic without intermittent negation.
Any logical reason to keep `isNoReplication()`?